File: Cpp/349_Intersection_of_Two_Arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        vector<int> result;
        unordered_set<int> s(nums1.begin(), nums1.end());

        for(auto& num : nums2){
            if(s.count(num) == 0) continue;
            s.erase(num);
            result.push_back(num);
        }
        
        return result;
    }
};

File: Kotlin/349_Intersection_of_Two_Arrays.kt
class Solution {
    fun intersection(nums1: IntArray, nums2: IntArray): IntArray {
        var result = mutableListOf<Int>()
        var s1 = nums1.toMutableSet()
        var s2 = nums2.toMutableSet()
        
        s1.forEach {
            if(s2.contains(it)){
                result.add(it)
            }
        }
        
        return result.toIntArray()
    }
}